### PR TITLE
fix: error generation format (#23)

### PR DESCRIPTION
**Pull request by vladisavvv**

### CODE REVIEW

File: aidial_sdk/application.py [hunk 1]
```diff
@@ -114,3 +116,9 @@ async def _chat_completion(
 
             log_debug(f"response: {response_body}")
             return JSONResponse(content=response_body)
+
+    def _exception_handler(self, request: Request, exc: HTTPException):
```
Comment (adubovik) [inline aidial_sdk/application.py]: `@staticmethod`

File: tests/applications/runtime_broken_application.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        self, request: Request, response: Response
+    ) -> None:
+        response.set_response_id("test_id")
+        response.set_created(0)
+
+        with response.create_single_choice() as choice:
+            choice.append_content("Test content")
+            await response.aflush()
+
+            raise HTTPException("Test error", 503)
```
Comment (adubovik) [inline tests/applications/runtime_broken_application.py]: It's worth adding applications which throw other types of exceptions: HTTPException from FastAPI, ValueError, Exception etc.

File: tests/test_errors.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    response = test_app.post(
+        "/openai/deployments/test_app/chat/completions",
+        json={
+            "messages": [{"role": "user", "content": "Test content"}],
+            "stream": True,
+        },
+        headers={"Api-Key": "TEST_API_KEY"},
+    )
+
+    for index, value in enumerate(response.iter_lines()):
```
Comment (adubovik) [inline tests/test_errors.py:106]: Let's add a helper to deal with streams in tests.
`tests/utils/stream.py: def parse_stream(response) -> List[str]`
which automatically asserts empty lines and returns a list of data chunks without `data: ` prefix.

**Review by adubovik — CHANGES_REQUESTED** [review]

File: tests/test_errors.py (hunk 3 above)
Comment (vladisavvv) [reply tests/test_errors.py:106]: If you don't mind, I suggest doing this as part of https://github.com/epam/ai-dial-sdk/issues/12 (global unit test issue)

File: aidial_sdk/application.py (hunk 1 above)
Comment (vladisavvv) [reply aidial_sdk/application.py]: fixed

File: tests/applications/runtime_broken_application.py (hunk 2 above)
Comment (vladisavvv) [reply tests/applications/runtime_broken_application.py]: added more tests

**Review by adubovik — APPROVED** [review]

**Pull request merged**